torch_geometric/nn/conv/message_passing.py

```diff
@@ -353,6 +356,59 @@ def propagate(self, edge_index: Adj, size: Size = None, **kwargs):
 
         return out
 
+    def edge_update(self, edge_index: Adj, size: Size = None, **kwargs):
+        r"""The call to generate edge updates from messages.
+
+        Args:
+            edge_index (Tensor or SparseTensor): A :obj:`torch.LongTensor` or a
+                :obj:`torch_sparse.SparseTensor` that defines the underlying
+                graph connectivity/message passing flow.
+                See :meth:`propagate` for more information.
+            size (tuple, optional): The size :obj:`(N, M)` of the assignment
+                matrix.
+                See :meth:`propagate` for more information.
+            **kwargs: Any additional data which is needed to update edges featres.
+        """
+        decomposed_layers = 1 if self.__explain__ else self.decomposed_layers
+        for hook in self._edge_update_forward_pre_hooks.values():
+            res = hook(self, (edge_index, size, kwargs))
+            if res is not None:
+                edge_index, size, kwargs = res
+
+        size = self.__check_input__(edge_index, size)
+
+        if decomposed_layers > 1:
+            user_args = self.__user_args__
+            decomp_args = {a[:-2] for a in user_args if a[-2:] == '_j'}
+            decomp_kwargs = {
+                a: kwargs[a].chunk(decomposed_layers, -1)
+                for a in decomp_args
+            }
+            decomp_out = []
+
+        for i in range(decomposed_layers):
+            if decomposed_layers > 1:
+                for arg in decomp_args:
+                    kwargs[arg] = decomp_kwargs[arg][i]
+
+            coll_dict = self.__collect__(self.__user_args__, edge_index,
```

Comment:
I think this is problematic, as `self.__collect__` will collect all arguments (even for those in `propagate` although we only need the arguments passed to `edge_message`. I think we need to disentangle `__user_args__` in arguments for `propagate` and arguments for `edge_message`.

Reply:
understood, I will figure out how to clearly do this